plugins/Services/plugin.py

```diff
@@ -66,10 +66,16 @@ def __init__(self, irc):
         self.reset()
 
     def reset(self):
-        self.channels = []
-        self.sentGhost = None
-        self.identified = False
-        self.waitingJoins = {}
+        self.state = {}
+
+    def _getState(self, irc):
+        class State(object):
+            def __init__(self):
+                self.channels = []
+                self.sentGhost = None
+                self.identified = False
+                self.waitingJoins = []
```

Comment:
Let's not create a class on every call... define it at the toplevel.
And no need to inherit from `object`

Reply:
Okay and okay.

Reply:
Done and done.